### PR TITLE
CNV-89223: Add clickable NAD links in VM network details tab

### PR DESCRIPTION
**Pull request by sjd78**
## 📝 Description

Replace plain-text NAD names in the VM details Configuration > Network tab with clickable `ResourceLink` components that navigate to the NetworkAttachmentDefinition detail page.

### Changes
- **NetworkCell**: New component that renders `ResourceLink` for multus networks, parsing `networkName` in `namespace/name` or bare `name` format
- **Pod networking**: Shown as plain text (no link) since pod networks don't reference external resources
- **Graceful fallback**: Malformed `networkName` values (empty name/namespace) fall back to plain text
- **Type update**: Added `pod?: object` to `SimpleNICPresentation.network` for proper pod network detection

## 🔗 Links
- JIRA: [CNV-89223](https://issues.redhat.com/browse/CNV-89223)
- Parent epic: [CNV-89211](https://issues.redhat.com/browse/CNV-89211)

## ✅ Verification
- [x] TypeScript compiles cleanly
- [x] ESLint passes (including max-lines rule)
- [x] Pre-commit hooks pass
- [x] Multi-perspective review (UX, QE, Security, KubeVirt Expert) — no blockers

## 📹 Demo
The "Network" column in the VM Configuration > Network tab now shows clickable links that navigate to the NAD detail page. Pod networking entries remain as plain text.

Made with [Cursor](https://cursor.com)

## Summary by CodeRabbit

- **New Features**
  - Improved NIC network display in virtual machine configuration tables.
  - When a Multus network name is available, it now links to the corresponding network attachment resource.
  - Supports network names with or without namespaces, with sensible fallbacks for malformed or missing values.
  - Shows pod networking when present; otherwise shows a “no data” placeholder.
- **Tests**
  - Added unit tests covering network-name parsing and edge cases.

### CODE REVIEW

File: src/views/virtualmachines/details/tabs/configuration/network/components/list/NetworkCell.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+): { name: string; namespace: string } => {
+  const slashIndex = networkName.indexOf('/');
+  if (slashIndex === -1) {
+    return { name: networkName, namespace: vmNamespace };
+  }
+  return {
+    name: networkName.substring(slashIndex + 1),
+    namespace: networkName.substring(0, slashIndex),
+  };
+};
```
Comment (upalatucci) [inline src/views/virtualmachines/details/tabs/configuration/network/components/list/NetworkCell.tsx]: can we have this function in a general utils folder? Like in `src/utils/resources/nad` or something like that? And we can call it like `parseVMMultusIntoNAD`

**Review by upalatucci — APPROVED** [review]

File: src/views/virtualmachines/details/tabs/configuration/network/components/list/NetworkCell.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (sjd78) [reply src/views/virtualmachines/details/tabs/configuration/network/components/list/NetworkCell.tsx]: Since the namespace and the network names are not allowed to have a "/" character, this isn't an issue.  unit tests added to show the various outcomes
